monero-wallet-generator.html

```diff
@@ -11183,31 +11186,47 @@ <h5 id="gpg-instructions">How to verify GPG signatures</h5>
   enableElement("lang_" + code, enable)
 }
 
+var cnUtil;
+var current_lang;
 function setCoin(index)
 {
   var enable
-  var language
   prefix_widget = document.getElementById("prefix_widget");
   if (index == 0) {
     cnUtil = cnUtilGen(moneroConfig);
     prefix_widget.value = "4";
-    enable = true
-    language = "english"
+    enable = true;
+    current_lang = 'english';
   }
   else if (index == 1) {
     cnUtil = cnUtilGen(aeonConfig);
-    prefix_widget.value = "W";
-    enable = false
-    language = "electrum"
+    prefix_widget.value = "Wm";
+    enable = false;
+    current_lang = 'electrum';
   }
   enableLanguage("en", enable)
   enableLanguage("es", enable)
   enableLanguage("pt", enable)
   enableLanguage("jp", enable)
   enableElement("show_qr_code", enable)
-  genwallet(language);
 }
 
+var oldSelectionIndex = false;
+function restore_old_selection_index()
+{
+  if (oldSelectionIndex && document.getElementById("currencyDropDownList").value == "Monero") {
+    document.getElementById("currencyDropDownList").value = "Aeon";
+    setCoin(1);
```

Comment:
Can you explain how we can get in that situation ? I don't understand this.

Reply:
on the actual master branch, when you select a new value on the drop-down list, a confirmation box 'Are you sure? This wallet cannot be recovered once a new wallet is generated.' is displayed. if you click OK, all is fine, a new wallet is generated according to the selected option in the drop-down list.
but if you cancel, the wallet is not generated and remains unchanged (as expected) the drop-down list is now on the new value. drop-down list and wallet does not match any more.
so i decided to use the variable 'oldSelectionIndex' like a flag. 'oldSelectionIndex' is set to 'false' most of the time. but i set it to 'true' is the drop-down must be reverted to former value. (when the user click on cancel after a new drop-down choice)
the restore_old_selection_index() function do the revert if 'oldSelectionIndex' is true then set the page with correct conf (setcoin(), according to the revert drop-down list) and reset 'oldSelectionIndex' to false.
so now, when a user select a new option on drop-down list, (line 10728), 'setcoin()' is executed then genwallet() is called. a confirmation box appears: if the user click on OK, genwallet() is executed and restore_old_selection_index() do nothing because 'oldSelectionIndex' is false. if the user click on cancel, genwallet() is dropped and 'oldSelectionIndex' is set to true. restore_old_selection_index() revert the drop-down list option and the page conf (setcoin()).